lib/chai/core/assertions.js

```diff
@@ -1501,31 +1501,108 @@ module.exports = function (chai, _) {
   Assertion.addMethod('key', assertKeys);
 
   /**
-   * ### .throw(constructor)
-   *
-   * Asserts that the target function will throw a specific error, or specific type of error
-   * (as determined using `instanceof`), optionally with a RegExp or String inclusion test
-   * for the error's message.
-   * If an Error instance is provided, it asserts that the error thrown and the instance
-   * provided are the same.
-   * If an Error constructor is provided, it asserts that the error thrown is an instance
-   * of the constructor provided.
-   *
-   *     var err = new ReferenceError('This is a bad function.');
-   *     var fn = function () { throw err; }
-   *     expect(fn).to.throw();
-   *     expect(fn).to.throw(ReferenceError);
-   *     expect(fn).to.throw(Error);
-   *     expect(fn).to.throw(/bad function/);
-   *     expect(fn).to.not.throw('good function');
-   *     expect(fn).to.throw(ReferenceError, /bad function/);
-   *     expect(fn).to.throw(err);
-   *
-   * Furthermore, `throw` changes the context of the assertion (just like `property`)
-   * to the thrown error. This permits for further chainable assertions on the thrown error.
-   *
-   *     expect(fn).to.throw(ReferenceError)
-   *       .and.have.property('message').equal('This is a bad function.');
+   * ### .throw([errorLike], [errMsgMatcher])
+   *
+   * Invokes the target function and asserts that an error is thrown, optionally
+   * asserting that the thrown error is an instance of the given error
+   * constructor, or that the thrown error is referentially equal to the given
+   * error instance, and/or that the thrown error's message contains a given
+   * string or matches a given regular expression.
```

Comment:
This is quite a chunky sentence, I'd like to simplify this into a series of shorter statements, maybe even bullets? Maybe something like...

> Invokes the target function, expecting it to throw an error. Passing in an Error constructor as the first argument will assert that the thrown error should match. Passing in an Error instance (`new Error('...')`) will assert that both errors should be referentially equal (`===`). Passing a string will assert that the thrown error's message matches the given string. Passing a regular expression will assert that the thrown error's message matches the given regular expression.